embassy-macros/src/macros/cortex_m_interrupt_declare.rs

```diff
@@ -9,8 +9,7 @@ pub fn run(name: syn::Ident) -> Result<TokenStream, TokenStream> {
     let result = quote! {
         #[allow(non_camel_case_types)]
         pub struct #name_interrupt(());
-        unsafe impl ::embassy::interrupt::Interrupt for #name_interrupt {
-            type Priority = crate::interrupt::Priority;
+        unsafe impl ::embassy_cortex_m::interrupt::Interrupt for #name_interrupt {
```

Comment:
Should these macros be moved into the embassy-cortex-m crate? Feels a little odd having them in here given that only `cortex-m` chips will ever use them.

Reply:
I kept them here because I didn't want to make yet another `embassy-cortex-m-macros` crate. :sweat_smile: 

Either way, it's not "public API", users aren't supposed to use `embassy-macros` directly, but use the reexports from the other crates.

Reply:
Ah yes, I forgot that proc-macros need to be in their own crate. I wonder if that limitation will be lifted at some point.